content/ytt/docs/latest/how-to-write-schema.md

```diff
@@ -0,0 +1,238 @@
+  name: ""
+```
+
+
+
+---
+## Specific Use-Cases
+
+A few less common, but real-world scenarios:
+
+- [setting a default value for arrays](#setting-a-default-value-for-an-array)
+- [marking a Data Value as optional](#marking-a-data-value-as-optional)
+- [allowing multiple types of maps or arrays](#allowing-multiple-types-of-maps-or-arrays)
+- [declaring "pass-through" Data Values](#declaring-pass-through-data-values)
+
+### Setting a Default Value for an Array
+
+As explained in [Data Values Schema Reference: Defaults for Arrays](lang-ref-ytt-schema.md#defaults-for-arrays), the default value for an array is an empty list (i.e. `[]`).
+
+In some cases, it is useful to provide a non-empty default value for an array. To do so, one typically uses a Data Values overlay.
+
+For example, with this schema:
+
+```yaml
+#@data/values-schema
+---
+app_domains:
+- ""
+```
+
+and this Data Values overlay:
+
+```yaml
+#@data/values
+---
+app_domains:
+- apps.cf-apps.io
+- mobile.cf-apps.io
+```
+
+The default value for `app_domains` will be `["apps.cf-apps.io", "mobile.cf-apps.io"]`.
+
+### Marking a Data Value as Optional
+
+Sometimes, it can be useful to define a section of Data Values as optional. This typically means that templates that rely on those values conditionally include output that use the contained value.
+
+For example, the following template:
+```yaml
+#@ load("@ytt:data", "data")
+---
+...
+spec:
+  #@ if data.values.load_balancer:
+  loadBalancerIP: #@ data.values.load_balancer.static_ip
+  #@ end
+...
+```
+will only include `spec.loadBalancerIP` if a value is provided for the `load_balancer` Data Value.
+
+One notes this in Schema using the `@schema/nullable` annotation:
+
+```yaml
+#@data/values-schema
+---
+#@schema/nullable
+load_balancer:
+  static_ip: ""
+```
+
+which indicates that `load_balancer` is `null`, by default. However, if a value _is_ provided for `load_balancer`, it must be the `static_ip` and have a value that is a **string**.
+
+For more details see [Data Values Schema Reference: `@schema/nullable`](lang-ref-ytt-schema.md#schemanullable).
+
+### Allowing Multiple Types of Maps or Arrays
+
+In rare cases, a given Data Value needs allow more than one type.
+
+Currently, `ytt` Schema does not explicitly support specifying more than one type for a Data Value.
+
+In the meantime, one can mark such types as having any type:
+
+```yaml
+#@schema/type any=True
+int_or_string: ""
+```
+so that:
+- `int_or_string` is, by default, an empty string
+- it can accept an **integer** or a **string** ... or any other type.
+
+If it is critical to ensure that the type of `int_or_string` to be _only_ an integer or string, one can include a validating library that does so explicitly:
+
+```python
+load("@ytt:assert", "assert")
+load("@ytt:data", "data")
+
+if type(data.value.int_or_string) not in ("int", "string"):
+  assert.fail("'int_or_string' must be either an integer or a string.")
+end
+```
+
+### Declaring "Pass-through" Data Values
```

Comment:
I read this under the headings Setting Default Values > Specific Use Cases, and expected to learn how to solve this use case in this paragraph but I only see a statement of the use case. I think the answer for it is to use `schema/type any=True`, It may be useful to say that here.

Reply:
Yup, absolutely forgot to flesh out that use-case. Great catch, @cari-lynn!

Will complete.